woocommerce/woocommerce-accommodation-bookings
# Fix/201 Separate Calendar Design and Availability for Check-in and Check-out Times!

### All Submissions:






### Changes proposed in this Pull Request:

**IMP**: This fix requires changes in the bookings plugin, so the fix will work only when the booking PR https://github.com/woocommerce/woocommerce-bookings/pull/3378 is also used with this PR.

- This PR mainly fixes the issue “fully booked days show as partially booked” as mentioned in #201.
- Introduces separate availability for check-in and check-out days.
- (Scenario before this PR:) For example, the 10th and 11th (1 night) are booked, both days are kept green, and hovering over them shows the days are partially booked. The problem was, that the 10th should not be available (RED) for Check-ins, but it should ALSO be partially booked (GREEN) for Check-outs as someone still can check out on that day!
- This PR fixes that issue by showing the 10th as red (fully booked) when someone is selecting check-in day, and as soon as, the check-in day is selected, the 10th is converted to green!
- There are now 3 types of fully booked days:
    - Fully booked for Check-in days
    - Fully booked for Check-out days
    - Fully booked for days (when a day is fully booked for check-ins and check-outs, both)
- Hovering over “Fully booked for Check-in days”, we’ll see `Available for Check-out only.`

_The 12th is fully booked for check-ins._
![image](https://user-images.githubusercontent.com/25176325/180998070-7aea7ba6-fc4a-4083-937e-eec06b1f1364.png)
[image link](https://www.screencast.com/t/kuUsHqcGq)

- Hovering over “Fully booked for Check-out days”, we’ll see `Available for Check-in only.`

_The 18th is fully booked for check-outs._
![image](https://user-images.githubusercontent.com/25176325/180998456-f3e42f20-f5db-4a9d-96f0-e6a8aa8a824b.png)
[image link](https://www.screencast.com/t/1BhYfl3KU)

- It also shows `SELECT CHECK-IN` / `SELECT CHECK-OUT` / `SELECTED! RE-SELECT TO CHANGE YOUR CHECK-IN` as per the selection made by the user.

![acom-201-1](https://user-images.githubusercontent.com/25176325/181000818-0322bcdd-b0db-4239-a0f1-9682d602e10e.gif)
[image link](https://www.screencast.com/t/Xl5oWSnklvx)

- Issue #231 is also fixed because we are not using the triangle display and its CSS is also completely removed in this PR.
- Issue #235 is also fixed because now we will see fully booked days separately for check-ins and check-outs, which will end the confusion.
- Issue #243 is also fixed because the fully booked check-in day is still selectable for check-outs.
- Issue #257 is also fixed because the check-in days will not be seen as partially booked.
- Issue #298 is also fixed because the fully booked days are now turned red.

Closes #201,
Closes #218,
Closes #231,
Closes #235,
Closes #243,
Closes #257,
Closes #298.

**Question:** When check-in day is selected, why not make the previous days white/not selectable? And why not keep green only the days till the next not-available/red day? Means, why not shrink down the range?
**Answer:** In the current behavior, after check-in and checkout days are selected, a user can still select a new check-in day anywhere on the calendar, which is more user-friendly than shrinking down the calendar days.

### How to test the changes in this Pull Request:

1. Import [booking-product-4128-2022-07-26.zip](https://github.com/woocommerce/woocommerce-accommodation-bookings/files/9189554/booking-product-4128-2022-07-26.zip).
2. The product has 2 resources. The resource “201-res-1” has maximum availability per block is 1. And “201-res-2” has maximum availability per block is 2.
3. At the Front end, see the text on the top of the calendar, “SELECT CHECK-IN”.
4. Select 17th (for example).
5. See the top text changed to “SELECT CHECK-OUT”.
6. Select 18th and book.
7. Now the 17th should be RED and not available to select as a check-in day.
8. Select 15th, see the 17th is now green; as it can be selected as a check-out day. Also, see the 18th is turned red from green; as it is not available to be selected as a check-out day.
9. Try to make bookings with resource 2, which has 2 quantities per block. It should turn the days red only when 2 bookings are made on a single block.
10. Also, check if the fix works when:
    1. Resources are automatically assigned.
    2. No resources used at all.
    3. No resources + Multiple quantities can be booked at the product level.
    4. With/out Persons.
    5. Any type of possible combination in the setting.
11. Please make sure that a normal booking product (not accommodation) is not affected in any way.
12. Also please check each connected issue ticket is resolved by this PR and see if the issues mentioned there are fixed.

**Note for QA:** This PR needs to be checked with https://github.com/woocommerce/woocommerce-bookings/pull/3352 too because it auto-selects the previously selected days when the resource is switched, so we need to make sure this PR works fine with it. This can be tested when all PRs are merged in the testing branch.

### Other information:




### Changelog entry

> Fix - Fully booked days show as partially booked - Day after booking shows partially booked.

### Release Todo Items
- Generate a pot file with `WP CLI`.
